MIDI Editor/rig-trimCCs.eel

```diff
@@ -0,0 +1,532 @@
+            ) : mouse_cap & 0x04 ? (    // CTRL key depressed
+                trim_requested = 5;
+            );
+        );
+    );
+    
+    trim_requested;
+);
+
+
+function run()  local(trim_requested, trimmed_cnt, c)
+(
+    set_default_colors();
+    gfx_a = 1;
+
+    draw_end_x = gfx_w - 22;
+    draw_end_y = gfx_h - 80;
+    gfx_x = draw_start_x;
+    
+    gfx_y = draw_start_y;
+
+    center_x = floor(draw_start_x + (draw_end_x - draw_start_x) / 2 + 0.5);
+    center_y = floor(draw_start_y + (draw_end_y - draw_start_y) / 2 + 0.5);
+
+    gfx_w != last_w ? (
+        center_x = floor(draw_start_x + (draw_end_x - draw_start_x) / 2 + 0.5);
+        slider_last_x = center_x;
+        last_w = gfx_w;
+    );
+
+    // Check if lmb down and mouse cursor outside window
+    mouse_cap >= 1 && (mouse_x <= 0 || mouse_x >= gfx_w || mouse_y < 2 || mouse_y >= gfx_h) ? (
+        lmb_click_outside_window = true;
+    ) : mouse_cap == 0 ? (
+            lmb_click_outside_window = false;
+    );
+
+    check_cc_lane();
+    
+    gfx_x = center_x - floor(s_w_trim / 2 + 0.5);   // centered
+    //gfx_x = draw_start_x;                         // left aligned
+    trim_requested = check_trim_btn(gfx_x, draw_start_y + (6 * gfx_texth), s_w_trim, s_h_trim, 12);
+
+    trimmed_cnt = trim(trim_requested);  //  Do it!
+
+    // add "undo point" if necessary:
+    trimmed_cnt > 0 ? (
+        Undo_OnStateChange(sprintf(#, "Trim CC%d events (x%d)", last_clicked_cc_lane, trimmed_cnt));
+        undo_cnt += 1;
+    );
+
+    c = gfx_getchar();
+    
+    // ESC: exit
+    c == 27 ? gfx_quit();
+    
+    // ctrl+Z: Undo
+    c == 26  &&  ((mouse_cap & 0x38)==0) ? (
+    
+      #undo_desc = "";
+      Undo_CanUndo2(#undo_desc, 0) ? (                   
+          Undo_DoUndo2(0);	// -> non zero if done
+          //Main_OnCommand(40029, 0); // undo
+
+            // If this was one of OUR undos then restore our state as well.
+          (strncmp(#undo_desc, "Trim CC", 7) == 0)  &&  (undo_cnt > 0) ? (
+                last_after_trim_checksum = stack_pop();
+                last_unselected_epsilon = stack_pop();
+                epsilon = stack_pop();
+                undo_cnt -= 1;
+            );
+         );
+    );
+
+    mouse_cap == 0 ? (
+        lmb_down = false;
+    );
+
+    last_h = gfx_h;
+    last_w = gfx_w;
+    gfx_update();
+    
+    // Loop unless graphics windows closed: 
+    c != -1 ? (
+        defer("run();");
+    );
+);
+
+
+function init()
+(
+    gfx_init(SCRIPT_NAME, 230, 170);      // show script window
+    
+    // Attach "pin on top" button (if Reaper 6.24+ with JS extension installed):
+    // FIXME: works randomly when script launched from toolbar!!
+    //GetAppVersion(#reaper_version);
+    //strnicmp(#reaper_version, "6.24", 4) >=0  &&  APIExists("JS_Window_Find")  ? (
+    (APIExists("JS_Window_Find") && APIExists("JS_Window_AttachTopmostPin"))  ? (
+        JS_Window_AttachTopmostPin(JS_Window_Find(SCRIPT_NAME, true));
```

Comment:
Sorry for the long delay. This fails to parse if js_ReaScript API is not installed. This should use the `extension_api("JS_Window_...", ...)` syntax instead.

If the name of the script is just "Trim CCs", then the filename should be "rig_Trim CCs.eel" and `@description` should be "Trim CCs" to match.

The documentation mention adding the script to the MIDI Editor section, but the package is configured to add the script to the Main section only. Consider registering the script to the MIDI Editor section or both sections.